packages/hardhat-chai-matchers/src/internal/emit.ts

```diff
@@ -40,42 +41,48 @@ export function supportEmit(
   Assertion.addMethod(
     "emit",
     function (this: any, contract: Contract, eventName: string) {
+      // see buildAssert's jsdoc
```

Comment:
I'd like to see a one-line explanation here, something like `capture negated flag before async code executes. see buildAssert's jsdoc.`